docs/source/daml/reference/contract-keys.rst

```diff
@@ -62,6 +62,17 @@ Like ``fetch``, ``fetchByKey`` needs to be authorized by at least one stakeholde
 
 This means that if it fails, it doesn't guarantee that a contract with that key doesn't exist, just that the submitting Party doesn't know about it, or there are issues with authorization.
 
+.. _doescontractexistforkey:
+
+canSeeContractWithKey
+=====================
+
+``boolean <- canSeeContractWithKey contractKey``
+
+Use ``canSeeContractWithKey`` to check whether, in the current context, you can see an active contract for the current key. If the contract exists and the current context has the permission to see it, returns `True`, otherwise returns `False`.
+
+``canSeeContractWithKey`` needs to be authorized by **all** maintainers of the contract you are trying to lookup. The reason for that is denial of service protection. Without this restriction, a malicious Party could legitimately request another Party to validate any number of negative contract key lookups without that other Party ever having signed anything.
```

Comment:
As per the forum discussion, it's not just DoS, but privacy protection as well. Without these rules, you could poll contract keys without authorization to gain information

Reply:
I've substantially rewritten this in a way that explains why the authorization is needed for the system to work, rather than what could go wrong if we did something else. I think that's better, but open to the idea that I may be wrong here.